nrf_rpc/include/nrf_rpc.h

```diff
@@ -189,6 +209,65 @@ struct nrf_rpc_err_report {
 		.strid = _strid,					         \
 		.transport = _transport,                                         \
 		.err_handler = _err_handler,				         \
+		.bound_handler = NULL,					         \
+		.flags = NRF_RPC_FLAGS_WAIT_ON_INIT | NRF_RPC_FLAGS_INITIATOR,   \
+	}
+
+/** @brief Define a non-blocking group of commands and events.
+ *
+ * The NOWAIT group does not block the @ref nrf_rpc_init until binding completion.
+ * When the NOWAIT group is bound, the @ref nrf_rpc_group_bound_handler_t will
+ * be called. The NOWAIT group can exist in two roles: initiator and follower.
+ * The initiator initiates the endpoint binding. The follower waits for the initiator
+ * to bind the group. Both peers can be initiators, but there must always be at least
+ * one on either side of the IPC channel.
+ *
+ * @param _name          Symbol name of the group.
+ * @param _strid         String containing unique identifier of the group. Naming
+ *                       conventions the same as C symbol name. Groups on local
+ *                       and remote must have the same unique identifier.
+ * @param _transport     Group transport. It is used by group to communicate with
+ *                       a remote processor.
+ * @param _ack_handler   Handler of type @ref nrf_rpc_ack_handler_t called when
+ *                       ACK was received after event completion. Can be NULL if
+ *                       group does not want to receive ACK notifications.
+ * @param _ack_data      Opaque pointer for the `_ack_handler`.
+ * @param _err_handler   Handler of type @ref nrf_rpc_err_handler_t called when
+ *                       error occurred in context of this group. Can be NULL if
+ *                       group does not want to receive error notifications.
+ * @param _bound_handler Handler of type @ref nrf_rpc_group_bound_handler_t called
+ *                       when the group was successfuly bound. The callback is called
+ *                       each time the remote peer binds to the group. This can be used
+ *                       to detect remote peer reset and can be used by the application
+ *                       to reset the local state.
+ * @param _initiator     The group is the initiator.
+ */
+#define NRF_RPC_GROUP_DEFINE_NOWAIT(_name, _strid, _transport, _ack_handler,      \
```

Comment:
This macro looks almost the same like NRF_RPC_GROUP_DEFINE You can consider simplification by creating, local macro and use it under these two. For example NRF_RPC_GROUP_DEFINE_INTERNAL__()

Reply:
Not sure, then the user could use the internal macro to create a group that is neither `NOWAIT` or normal.

Reply:
I mean to keep both macros and use internal macro under them to avoid duplication in macro body

Reply:
I think when user use something which has internal in name then he does it on his own responsibility

Reply:
Done as suggested.